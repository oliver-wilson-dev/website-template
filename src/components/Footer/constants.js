export const GITHUB_PROFILE_URL = 'https://github.com/oliver-wilson-dev/';
export const PHONE_NUMBER = '07580081310';
export const EMAIL_ADDRESS = 'contact@oliverwilson.dev';
