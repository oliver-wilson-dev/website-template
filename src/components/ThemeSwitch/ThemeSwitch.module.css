.moon {
    width: 30px;
    height: 30px;
}

.sun {
    width: 30px;
    height: 30px;
}

.btn {
    background-color: unset;
    border: none;
    padding: 0px;
    opacity: 1;

    transition: transform 0.1s;
    transition-timing-function: linear;
}

.hide {
    transform: scale(0);
}

.show {
    transform: scale(1);
}