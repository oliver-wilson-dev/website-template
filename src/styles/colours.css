
$background-grey: #e9ecef;

$dark-grey:  #333333;

$text-dark-grey:  #333333;
$text-light-grey:  #ffffff;

$link-blue: #1540cc;
$link-yellow: #fcbf49;

$nav-background: #f8f8f8;