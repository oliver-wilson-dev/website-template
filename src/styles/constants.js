export const LIGHT_THEME = Symbol('LIGHT_THEME');
export const DARK_THEME = Symbol('DARK_THEME');
