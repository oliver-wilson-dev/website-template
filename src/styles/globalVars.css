$max-content-width: 1000px;
$navigation-height: 50px;