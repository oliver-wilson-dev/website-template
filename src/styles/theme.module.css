@import "./colours.css";

.light {
    --background-colour: $background-grey;
    --background-colour-inverse: $dark-grey;
    --text-colour: $text-dark-grey;
    --text-colour-inverse: $text-light-grey;
    --side-menu-focus: $dark-grey;
}

.dark {
    --background-colour: $dark-grey;
    --background-colour-inverse: $background-grey;
    --text-colour: $text-light-grey;
    --text-colour-inverse: $text-dark-grey;
    --side-menu-focus: $background-grey;
}
