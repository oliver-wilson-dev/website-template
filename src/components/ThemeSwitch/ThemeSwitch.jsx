import React, { useState, useEffect } from 'react';
import cn from 'classnames';
import Moon from './assets/moon.svg';
import Sun from './assets/sun.svg';
import styles from './ThemeSwitch.module.css';
import { LIGHT_THEME } from '../../styles/constants';

const ThemeSwitch = ({ theme, toggleTheme, className }) => {
  const [hide, setHide] = useState(false);
  const [hideComplete, setHideComplete] = useState(false);

  useEffect(() => {
    if (hide && hideComplete) {
      setHide(false);
      setHideComplete(false);
      toggleTheme();
    }
  }, [theme, hide, hideComplete]);

  const onClick = () => {
    setHide(true);
  };

  const onTransitionEnd = () => {
    if (hide && !hideComplete) {
      setHideComplete(true);
    }
  };

  return (
    <button className={cn(styles.btn, { [styles.hide]: hide, [styles.show]: !hide }, className)} type="button" onClick={onClick} onTransitionEnd={onTransitionEnd}>
      {theme === LIGHT_THEME ? <Moon className={styles.moon} /> : <Sun className={styles.sun} />}
    </button>
  );
};

export default ThemeSwitch;
