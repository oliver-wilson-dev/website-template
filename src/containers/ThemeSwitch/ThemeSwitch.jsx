import { connect } from 'react-redux';
import ThemeSwitch from '../../components/ThemeSwitch';
import { toggleTheme } from '../../state/actions';
import { getTheme } from '../../state/selectors';

const mapStateToProps = (state) => ({
  theme: getTheme({ state }),
});

const mapDispatchToProps = (dispatch) => ({
  toggleTheme: () => dispatch(toggleTheme())
});

export default connect(mapStateToProps, mapDispatchToProps)(ThemeSwitch);
