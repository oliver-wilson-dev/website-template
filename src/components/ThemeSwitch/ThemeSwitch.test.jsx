import React from 'react';
import { mount } from 'enzyme';
import ThemeSwitch from './ThemeSwitch';
import { DARK_THEME, LIGHT_THEME } from '../../styles/constants';

jest.mock('./assets/sun.svg', () => {
  const Sun = () => null;

  return Sun;
});

jest.mock('./assets/moon.svg', () => {
  const Moon = () => null;

  return Moon;
});

const defaultProps = {
  theme: LIGHT_THEME,
  toggleTheme: jest.fn(),
  className: undefined
};

const render = (props = {}) => mount(<ThemeSwitch {...defaultProps} {...props} />);

describe('ThemeSwitch', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  describe('when the theme is the light theme', () => {
    it('should render correctly', () => {
      expect(render()).toMatchSnapshot();
    });

    describe('when providing an additional class', () => {
      it('should render correctly', () => {
        expect(render({ className: 'test-class' })).toMatchSnapshot();
      });
    });
  });

  describe('when the theme is the dark theme', () => {
    it('should render correctly', () => {
      expect(render({ theme: DARK_THEME })).toMatchSnapshot();
    });

    describe('when providing an additional class', () => {
      it('should render correctly', () => {
        expect(render({ className: 'test-class' })).toMatchSnapshot();
      });
    });
  });

  describe('when clicking the toggle theme button', () => {
    describe('and the transition has ended', () => {
      it('should call toggleTheme', () => {
        const component = render();

        component.find('button').simulate('click');
        component.find('button').simulate('transitionEnd');

        expect(defaultProps.toggleTheme).toHaveBeenCalled();
      });
    });

    describe('and the transition has not ended', () => {
      it('should call toggleTheme', () => {
        const component = render();

        component.find('button').simulate('click');

        expect(defaultProps.toggleTheme).not.toHaveBeenCalled();
      });
    });

    describe('and the transition has ended but the button should not be hidden', () => {
      it('should call toggleTheme', () => {
        const component = render();

        component.find('button').simulate('transitionEnd');

        expect(defaultProps.toggleTheme).not.toHaveBeenCalled();
      });
    });
  });
});
