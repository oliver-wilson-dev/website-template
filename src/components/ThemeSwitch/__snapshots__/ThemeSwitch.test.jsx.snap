// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitch when the theme is the dark theme should render correctly 1`] = `
<ThemeSwitch
  theme={Symbol(DARK_THEME)}
  toggleTheme={[MockFunction]}
>
  <button
    className="btn show"
    onClick={[Function]}
    onTransitionEnd={[Function]}
    type="button"
  >
    <Sun
      className="sun"
    />
  </button>
</ThemeSwitch>
`;

exports[`ThemeSwitch when the theme is the dark theme when providing an additional class should render correctly 1`] = `
<ThemeSwitch
  className="test-class"
  theme={Symbol(LIGHT_THEME)}
  toggleTheme={[MockFunction]}
>
  <button
    className="btn show test-class"
    onClick={[Function]}
    onTransitionEnd={[Function]}
    type="button"
  >
    <Moon
      className="moon"
    />
  </button>
</ThemeSwitch>
`;

exports[`ThemeSwitch when the theme is the light theme should render correctly 1`] = `
<ThemeSwitch
  theme={Symbol(LIGHT_THEME)}
  toggleTheme={[MockFunction]}
>
  <button
    className="btn show"
    onClick={[Function]}
    onTransitionEnd={[Function]}
    type="button"
  >
    <Moon
      className="moon"
    />
  </button>
</ThemeSwitch>
`;

exports[`ThemeSwitch when the theme is the light theme when providing an additional class should render correctly 1`] = `
<ThemeSwitch
  className="test-class"
  theme={Symbol(LIGHT_THEME)}
  toggleTheme={[MockFunction]}
>
  <button
    className="btn show test-class"
    onClick={[Function]}
    onTransitionEnd={[Function]}
    type="button"
  >
    <Moon
      className="moon"
    />
  </button>
</ThemeSwitch>
`;
